.cargo/config.toml

```diff
@@ -12,7 +12,74 @@ rustflags = [
 	# rustc additional warnings:
 	"--warn",
 	"unused_crate_dependencies",
-	# clippy additional warnings:
 	"--warn",
-	"clippy::wildcard_imports",
+	"rust_2018_idioms",
+	# clippy additional warnings:
+	# Suspicious
+	"-Wclippy::await_holding_lock",
+	# Restriction
+	"-Wclippy::dbg_macro",
+	"-Wclippy::exit",
+	"-Wclippy::float_cmp_const",
+	"-Wclippy::lossy_float_literal",
+	"-Wclippy::mem_forget",
+	"-Wclippy::rc_mutex",
+	"-Wclippy::rest_pat_in_fully_bound_structs",
+	"-Wclippy::string_add",
+	"-Wclippy::string_to_string",
+	"-Wclippy::todo",
+	"-Wclippy::verbose_file_reads",
+	# Complexity
+	"-Wclippy::char_lit_as_u8",
+	"-Wclippy::checked_conversions",
+	"-Wclippy::map_flatten",
+	"-Wclippy::useless_transmute",
+	# Style
+	"-Wclippy::let_unit_value",
+	"-Wclippy::missing_enforced_import_renames",
+	"-Wclippy::needless_borrow",
+	# Pedantic
+	"-Wclippy::bool-to-int-with-if",
+	"-Wclippy::cast_lossless",
+	"-Wclippy::default_trait_access",
+	"-Wclippy::doc_markdown",
+	"-Wclippy::enum_glob_use",
+	"-Wclippy::expl_impl_clone_on_copy",
+	"-Wclippy::explicit_deref_methods",
+	"-Wclippy::explicit_into_iter_loop",
+	"-Wclippy::filter_map_next",
+	"-Wclippy::flat_map_option",
+	"-Wclippy::fn_params_excessive_bools",
+	"-Wclippy::inefficient_to_string",
+	"-Wclippy::invalid_upcast_comparisons",
+	"-Wclippy::items_after_statements",
+	"-Wclippy::large_digit_groups",
+	"-Wclippy::large_stack_arrays",
+	"-Wclippy::large_types_passed_by_value",
+	"-Wclippy::linkedlist",
```

Comment:
The list has few lints like this, which are probably never going to be triggered but which would be nice when warned against, so I've left them. Let me know if you think we should prune these from the list.

Reply:
I think being extra careful is good, as long as it doesn't degrade/slow down type checking too much.